marshmallow/fields.py

```diff
@@ -1139,11 +1147,91 @@ class Dict(Field):
         'invalid': 'Not a valid mapping type.'
     }
 
-    def _deserialize(self, value, attr, data):
-        if isinstance(value, collections.Mapping):
-            return value
+    def __init__(self, values=None, keys=None, **kwargs):
+        super(Dict, self).__init__(**kwargs)
+        if values is None:
+            self.value_container = None
+        elif isinstance(values, type):
+            if not issubclass(values, FieldABC):
+                raise ValueError('"values" must be a subclass of '
+                                 'marshmallow.base.FieldABC')
+            self.value_container = values()
+        else:
+            if not isinstance(values, FieldABC):
+                raise ValueError('"values" must be of type '
```

Comment:
"must be of type" -> "must be a subclass of"

Reply:
Actually, scratch that. It would be good if the message matched that for the List class though - refer to "The type of the values" and "The value instances" respectively